app/Observers/AudioObserver.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\Audio;
+use App\Models\SegmentBucket;
+use App\Traits\TranscodeAudio;
+use Illuminate\Filesystem\FilesystemManager;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemException;
+
+class AudioObserver
+{
+    use TranscodeAudio;
+
+    private Filesystem $disk;
+
+    public function __construct(FilesystemManager $fs)
+    {
+        /* @var Filesystem $disk */
+        $disk = $fs->disk('minio.segment');
+
+        $this->disk = $disk;
```

Comment:
The $disk declaration can be moved to ServiceProvider. Thus, the class would depend only on `Filesystem` variable, not on FilesystemManager. And since Filesystem instance seems like reusable, it should not be reinitiated in code.